Content.Server/Nutrition/EntitySystems/DrinkSystem.cs

```diff
@@ -123,51 +126,42 @@ public float TotalHydration(EntityUid uid, DrinkComponent? comp = null)
-        if (!IsEmpty(uid, component))
+        var empty = IsEmpty(uid, component);
+        var color = empty ? "gray" : "yellow";
+        // only show the Opened text if it can be opened
+        if (empty || canOpen)
         {
-            if (TryComp<ExaminableSolutionComponent>(uid, out var comp))
-            {
-                //provide exact measurement for beakers
-                args.Message.AddMarkup($" - {Loc.GetString("drink-component-on-examine-exact-volume", ("amount", DrinkVolume(uid, component)))}");
-            }
-            else
-            {
-                //general approximation
-                var remainingString = (int) _solutionContainer.PercentFull(uid) switch
-                {
-                    100 => "drink-component-on-examine-is-full",
-                    > 66 => "drink-component-on-examine-is-mostly-full",
-                    > 33 => HalfEmptyOrHalfFull(args),
-                    _ => "drink-component-on-examine-is-mostly-empty",
-                };
-                args.Message.AddMarkup($" - {Loc.GetString(remainingString)}");
-            }
+            var openedText =
+                Loc.GetString(empty ? "drink-component-on-examine-is-empty" : "drink-component-on-examine-is-opened");
+            args.Message.AddMarkup($"\n{Loc.GetString("drink-component-on-examine-details-text", ("colorName", color), ("text", openedText))}");
         }
-    }
-
-    private void SetOpen(EntityUid uid, bool opened = false, DrinkComponent? component = null)
-    {
-        if(!Resolve(uid, ref component))
-            return;
 
-        if (opened == component.Opened)
+        if (empty)
             return;
 
-        component.Opened = opened;
+        // go between Opened and Xu / put Xu on a new line
+        args.Message.AddMarkup(canOpen ? " - " : "\n");
 
-        if (!_solutionContainer.TryGetSolution(uid, component.SolutionName, out _))
-            return;
-
-        if (EntityManager.TryGetComponent<AppearanceComponent>(uid, out var appearance))
+        if (TryComp<ExaminableSolutionComponent>(uid, out var comp))
         {
-            _appearance.SetData(uid, DrinkCanStateVisual.Opened, opened, appearance);
+            //provide exact measurement for beakers
+            args.Message.AddMarkup(Loc.GetString("drink-component-on-examine-exact-volume", ("amount", DrinkVolume(uid, component))));
+        }
+        else
+        {
+            //general approximation
+            var remainingString = (int) _solutionContainer.PercentFull(uid) switch
+            {
+                100 => "drink-component-on-examine-is-full",
+                > 66 => "drink-component-on-examine-is-mostly-full",
+                > 33 => HalfEmptyOrHalfFull(args),
+                _ => "drink-component-on-examine-is-mostly-empty",
+            };
+            args.Message.AddMarkup(Loc.GetString(remainingString));
```

Comment:
Huh? this should also be handled by ExaminableSolution, what is going on with drinks

Reply:
idk it sussy but the locale ids are all in drink component